mern-ecommerce/controllers/user.js

```diff
@@ -125,3 +125,19 @@ exports.purchaseHistory = (req, res) => {
       res.json(orders);
     });
 };
+
+exports.deleteUser = (req, res) => {
+  
+  const { userId } = req.params;
+  User.findByIdAndRemove(userId, (err, user) => {
+    if (err || !user) {
```

Comment:
The condition `if (err || !user)` is used in multiple places in the code, but it's not consistently handled. In some cases, it returns a 400 error with a specific message, while in others, it simply returns a 400 error without a message.

To maintain consistency, it's better to create a separate function to handle this error condition and return a standardized error response. This function can be reused throughout the code.

For example, you can create a function like this:

```
const handleUserNotFoundError = (res) => {
  return res.status(400).json({
    error: 'User not found',
  });
};
```

Then, you can use this function in your code like this:

```
if (err || !user) {
  return handleUserNotFoundError(res);
}
```

This approach makes the code more maintainable and easier to understand.